修正１

KazuGeoge opened the pull request:
ご指摘頂いた部分の修正です。
よろしくお願いします！

KazuGeoge committed:
修正１

ご指摘頂いた部分の修正です。
よろしくお願いします！

KazuGeoge committed:
修正２

・delegateとプロトコルを使用してWebViewに遷移
・TableViewControllerをNSObjectに変更しクラス名も変更
・WebViewもオートレイアウトで設定

またTableViewのCellの値を入れる部分の条件文がおかしくTableViewに何も出ない状態でしたのでその修正。
を行いました。

KazuGeoge committed:
degate先のTableView.reloadDataでのエラー

issueに現在の進捗と問題について記述しました。
ご確認よろしくお願いします！

KazuGeoge committed:
Delegate, Protocol練習。お気に入り追加

@kazuteru
解決しました！
アドバイス頂いた通り
｀｀｀tablePage.delegate = self｀｀｀
こちらで渡していたので、このインスタンスをdelegate先で設定すればいいという事がしっかり理解できました！
見やすくするためにアドバイス頂いた通り別でfavDelegateを作りそれを設定する形にして見ました！
ありがとうございました！

KazuGeoge committed:
マルチバイトによるタイトル分割の修正とお気に入りセクションと通常セクションの重複の修正

kazuteru commented on RssTab/model/BaseContent.swift:
```diff
@@ -4,15 +4,11 @@
-        self.url = url
-    }
+struct BaseContent {
+    var title :  String
```
細かいですが変数名の後ろにスペースは不要ですね
var title:  String

KazuGeoge commented:
全体的にその形にしていたので修正します！
ありがとうございます！

kazuteru commented on RssTab/model/Const.swift:
```diff
@@ -0,0 +1,15 @@
+//  Copyright © 2018年 城島一輝. All rights reserved.
+
+struct Const {
+    static let tech = "https://news.yahoo.co.jp/pickup/computer/rss.xml"
```
Constは全部大文字で記載することが多いですね
tech → TECH

kazuteru commented on RssTab/model/FeedItem.swift:
```diff
@@ -4,11 +4,13 @@
-    var url: String!
+struct FeedItem {
+    var title: String
+    var url: String
```
変数名がurlなのに、Stringてのが気になりますね 🤔

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+//
+//  TableViewController.swift
```
ファイル名変えましょう...

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+    
+    var sectionTitle: [String] = ["お気に入り"]
+    var cellTitle : [String] = []
+    var cellURL : [String] = []
```
上では変数でurlが小文字でしたけど、ここではURLで大文字なのでいずれか揃えましょう

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+    var sectionTitle: [String] = ["お気に入り"]
+    var cellTitle : [String] = []
+    var cellURL : [String] = []
+    var feedItems = [FeedItem]()
```
ここも上の初期化がvar cellURL : [String] = []のように[]でやっているところで、下が[FeedItem]()となっているのでどちらか揃えましょう！

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+}
+
+class TableViewDataSouce: NSObject, UITableViewDataSource, UITableViewDelegate {
+    
```
内部だけで使う変数にはprivateつけましょう

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+    }
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        var cellCount: Int = 0
```
0で初期化してるので: Intは不要そうですね

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+    
+     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let mainStoryboard = UIStoryboard(name: "Main", bundle: nil)
+        let webViewDetailVC = mainStoryboard.instantiateViewController(withIdentifier: "WebViewDetailView") as! WebViewDetailViewController
```
可能な限りアンラップは使わないようにしましょう
as! → as?
nilな場合落ちるので

kazuteru commented on RssTab/viewController/New Group/TableViewController.swift:
```diff
@@ -0,0 +1,90 @@
+            let feedItem = self.favCellURL[indexPath.row]
+            webViewDetailVC.webUrl = feedItem
+            let title = feedItems(indexPath: indexPath)
+            webViewDetailVC.barTitle = title.textLabel!.text!
```
ここもアンラップ

kazuteru commented on RssTab/viewController/QiitaArticleViewController.swift:
```diff
@@ -0,0 +1,69 @@
+        view.addSubview(tableView)
+    }
+    
+    func loadAPI() {
```
中でしか使わない関数にもprivateつけましょう

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -8,35 +8,41 @@
+        
         webView.delegate = self
-        self.view.addSubview(webView)
         let url = webUrl
```
この変数代入不要そうな...

kazuteru commented on RssTab/viewController/YahooTechArticleViewController.swift:
```diff
@@ -0,0 +1,133 @@
+        }
+        tablePage.cellTitle = newCellTitle
+        
+        for URL1 in tablePage.cellURL {
```
変数名に1, 2,のような数字をつけるのはやめましょう

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -8,35 +8,41 @@
 
 import UIKit
 
+protocol favProtocol {
```
クラス名、変数名、プロトコルなど基本大文字始まりとなります

KazuGeoge committed:
現状

KazuGeoge committed:
クラス、ViewControllerなど似た処理をまとめる

一つのViewControllerで必要なインスタンスを生成し
、同じような処理をまとめました。
また注意する点として以下のことを意識しました。
・インデント
・if文、for文、関数の前後に改行
・Constは大文字
・self、型の明示(右辺で型が明確な時など)、！(アンラップ)は不要な所以外使わない
・CGFloatは整数のみで0.０としない
・クラス名、プロトコルは最初大文字
・クラス名は端的にする
・どんな処理なのか分かりやすい関数名をつける
・階層を意識したグループ分けをする
・親Viewと子Viewを相互参照しない形を意識する

ご確認よろしくお願いします！

kazuteru commented on Podfile:
```diff
@@ -5,4 +5,6 @@ use_frameworks!
 target 'RssTab' do
   pod 'Alamofire', '~> 4.5'
   pod 'SwiftyJSON'
+  pod 'SDWebImage/GIF', '~> 4.0'
```
SDWebImage導入いいすね！
ただ、画像読み込みならNukeかkingfisherのほうがいまは一般的かもです(別の目的なら検討です

kazuteru commented on RssTab/model/BaseContent.swift:
```diff
@@ -11,7 +11,7 @@ import SwiftyJSON
     var title :  String?
     var link : String?
-    var image: String
+    var image: String?
```
imageという変数名だとUIImageと勘違いされるので、imageUrlStringなどにしましょう（imageUrlでもURLクラスと勘違いされうるので）

kazuteru commented on RssTab/model/BaseContent.swift:
```diff
@@ -20,8 +20,8 @@ struct BaseContent {
-    self.title = json["title"].string
-    self.link = json["url"].string!
-    self.image = json["user"]["profile_image_url"].string!
+        self.title = json["title"].string
```
ナイスインデント 👍

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+
+    @IBOutlet weak var tableView: UITableView!
+    var sectionTitle = [""]
+    var feedURL = URL(string: "")
```
""のようなStringの空値で初期化する必要はなさそうですね

var feedURL: URL? のような形が一般的かと

kazuteru replied:
無駄な処理をするとメモリが圧縮されるという意識を持つといいかもです

kazuteru replied:
変数もクラス内で利用するものにはprivateをつけましょう

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        tableView.delegate = tableViewDataSouce
+        tableView.dataSource = tableViewDataSouce
+        tableView.register(UITableViewCell.self, forCellReuseIdentifier:"cell")
+        view.addSubview(tableView)
```
すでにtableviewてaddSubviewされてますよね..？

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        }
+    }
+    // パースされたデータを設定
+    func setXMLData(parsedCellTitle: [String], parsedCellURL: [String]) {
```
配列の変数・引数は複数形にしましょう
parsedCellTitle → parsedCellTitles

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+    // パースされたデータを設定
+    func setXMLData(parsedCellTitle: [String], parsedCellURL: [String]) {
+        tableViewDataSouce.cellTitle = parsedCellTitle
+        tableViewDataSouce.cellLink = parsedCellURL
```
tableView.reloadData()
なくて大丈夫？

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+    }
+    // WebViewへの遷移
+    func pushWebView(webView: UIViewController) {
+        self.navigationController?.pushViewController(webView, animated: true)
```
self不要ですね

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        tableView.reloadData()
+    }
+    // WebViewへの遷移
+    func pushWebView(webView: UIViewController) {
```
pushWebViewという名前ですが、webViewController以外のViewControllerでも遷移可能そうなので普通にpushViewControllerとかに名前変更した方がいいかもですね

kazuteru commented on RssTab/viewController/TabsViewController.swift:
```diff
@@ -15,22 +15,38 @@ class TabsViewController: UITabBarController {
+                    yahooInternational.feedURL = URL(string: Const.INTERNATIONAL)
+                    yahooInternational.isXML = true
+        
+                    if let yahooEconomyVC = mainStoryboard.instantiateViewController(withIdentifier: "ArticleView") as? ArticleViewController {
```
ネストが深くなってますね
var viewControllers = [] を用意して、そこにappendしていく形にするのはどうでしょうか？

kazuteru commented on RssTab/viewController/Views/ReadJson.swift:
```diff
@@ -0,0 +1,47 @@
+                    json.forEach { (_, json) in
+                        let content = BaseContent(json: json)
+                        self.contentList.append(content)
+                        self.cellTitle.append(content.title!)
```
アンラップ

kazuteru commented on RssTab/viewController/Views/ReadJson.swift:
```diff
@@ -0,0 +1,47 @@
+                        self.contentList.append(content)
+                        self.cellTitle.append(content.title!)
+                        self.cellLink.append(content.link!)
+                        self.decodedJsonData?.setJsonData(decodedCellTitle: self.cellTitle, decodedCellLink: self.cellLink)
```
setJsonDataてforEach内の利用であってます..？

kazuteru commented on RssTab/viewController/Views/ReadXML.swift:
```diff
@@ -8,79 +8,73 @@
-    var delegate : XMLParserDelegate?
+    private var isSplitTilte: Bool? = true
+    var parsedXMLData: ParsedXMLData?
+    var cellTitle: [String] = []
```
前も指摘しましたが、配列の初期化方法も統一しましょう
private var feedItems = [FeedItem]()

var cellTitle: [String] = []

の2パターンがありますね

kazuteru commented on RssTab/viewController/Views/ReadXML.swift:
```diff
@@ -8,79 +8,73 @@
         
-        if self.feedItems.count > 0 {
-            var lastItem = self.feedItems[self.feedItems.count - 1]
+        if feedItems.count > 0 {
```
empty的なメソッドがあったかと思います(0かどうかをチェックする)

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -0,0 +1,98 @@
+        // XMLファイルをパース
+        if isXML == true {
+            readXML.parsedXMLData = self
+            // feedURLはConstで設定のためnilの可能性はなし
```
ナイスコメント

KazuGeoge committed:
アラート、TableViewの編集、検索Viewの設置

WebViewでのお気に入りの際のアラートの設定、TableViewの編集機能をつける、検索Viewの設置を行いました。
検索ViewはスマートログだとAPIを飛ばす段階でURLの最後に&word + (エンコードした検索単語)
飛ばし絞られた検索結果を受け取る形でしたが今回使用しているYahooのXMLやQiitaのJsonファイルの絞ったAPIをリクエストして受け取る形が見つけられなかったので、今回はまず全てのデータを格納して配列から文字列検索する形にしました。
ご確認宜しくお願いします！

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -54,45 +54,50 @@ class ArticleViewController: UIViewController, TableProtocol, FavProtocol, Parse
+        tableViewDataSouce.favCellTitles.append(titleCell)
+        tableViewDataSouce.favCellURLs.append(webURL)
+        
+        for title in tableViewDataSouce.cellTitles {
```
filterとか使うともっとスマートにかけるかもですね

kazuteru commented on RssTab/viewController/SearchViewController.swift:
```diff
@@ -0,0 +1,150 @@
+        searchBar.setShowsCancelButton(true, animated: true)
+        
+        let searchBarBottomY = searchBar.frame.origin.y + searchBar.bounds.height - tableView.contentOffset.y
+        overlapTableView.frame = CGRect(x: 0, y: searchBarBottomY, width: view.bounds.height, height: view.bounds.height - searchBarBottomY)
```
ここらへんはautolayoutを変更したいところ

オフラインで話しましょうか

kazuteru commented on RssTab/viewController/SearchViewController.swift:
```diff
@@ -0,0 +1,150 @@
+        tableViewDataSouce.searchSection = ["検索結果"]
+
+        for title in resultTitles {
+            if title.lowercased().contains(searchWord!) {
```
!

kazuteru commented on RssTab/viewController/TabsViewController.swift:
```diff
@@ -13,40 +13,46 @@ class TabsViewController: UITabBarController {
-                    }
-                }
-            }
+        if let yahooEconomyVC = mainStoryboard.instantiateViewController(withIdentifier: "ArticleView") as? ArticleViewController {
```
かなりきれいにはなってますが、似た処理が複数書かれているのでもう少しまとめられそうですね

kazuteru commented on RssTab/viewController/Views/TableViewDataSouce.swift:
```diff
@@ -9,84 +9,139 @@
-            webViewDetailVC.webViewDelegate = favDelegate
-            tableDelegate?.pushWebView(webView: webViewDetailVC)
-            print("記事URL:\(feedItem)")
+            if isFromSearchView == true {
```
インデントがずれてる..？

kazuteru commented:
関数の間は一行(コメントがあった場合はコメント+一行)空けましょう

kazuteru commented:
かなり書き方がきれいになりましたね！とても読みやすくなりました
引き続き、このクオリティでコードを書けるようにしましょう

今後は細かい部分の質を良くしていく作業になるかと思います

KazuGeoge committed:
json検索結果受け取り途中

jsonの検索結果を受け取る途中の所までです。
modelのBaseContentでjsonを受け取りjsonをデコードするファイルでその受け取ったデータを配列に入れる形です。
ご確認よりしくお願いします！

KazuGeoge committed:
APIを飛ばして検索する。似た処理をまとめる

ATNDのAPIを使用しGETパラメータを使用し検索する。
TabBarControllerのViewControllerのインスタンス化はfor文を使用し似た処理をまとめました。

お気に入りした記事以外を配列に格納する処理はfillterを使用し短くまとめました。

あとは各種改行やインデントなどの修正を行いました。
ご確認よろしくお願いします！

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -39,65 +38,63 @@ class ArticleViewController: UIViewController, TableProtocol, FavProtocol, Parse
         tableView.delegate = tableViewDataSouce
         tableView.dataSource = tableViewDataSouce
-        tableView.register(UITableViewCell.self, forCellReuseIdentifier:"cell")
+        tableView.register(ListCell.self, forCellReuseIdentifier:"cell")
```
`:`の裏にスペース空けましょう

kazuteru commented on RssTab/viewController/ArticleViewController.swift:
```diff
@@ -39,65 +38,63 @@ class ArticleViewController: UIViewController, TableProtocol, FavProtocol, Parse
+        // 判定用にURLをString型に戻す
+        let URLString = webURL.absoluteString
+        // お気に入りされているタイトルを配列から外す
+        tableViewDataSouce.cellTitles = tableViewDataSouce.cellTitles.filter {$0 != titleCell}
```
ナイスfilter！

kazuteru commented on RssTab/viewController/TabsViewController.swift:
```diff
@@ -15,38 +15,37 @@ class TabsViewController: UITabBarController {
-            viewControllers.append(yahooTechVC)
+        for articleView in articleViews {
+            if let ArticleView = mainStoryboard.instantiateViewController(withIdentifier: "ArticleView") as? ArticleViewController {
+                
```
ここらへん、Structかenum使ってまとめるといい具合にできそうですね
少なくともswitch文を使うときれいになりそう

kazuteru commented on RssTab/viewController/Views/ReadJson.swift:
```diff
@@ -15,22 +15,20 @@ protocol DecodedJsonData {
     private var cellTitles: [String] = []
     private var cellLinks: [String] = []
-    private var content: BaseContent?
     private var cellnikNames: [String] = []
```
`cellNickNames`ですかね

KazuGeoge committed:
webViewをWKWebViewに変更

webViewをWKWebViewに変更し、webview画面で戻る、進む、リロード、safariに飛ぶ、四つのボタンを設定しました。
またTabViewControllerではswich文を使いまとめましたが列挙やstructを上手くしよう出来ていない状況です。また設置したボタン自体にオートレイアウトがかけられず機種によってずれてしまっています。

kazuteru commented on RssTab.xcodeproj/project.pbxproj:
```diff
@@ -525,7 +525,7 @@
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
-				IPHONEOS_DEPLOYMENT_TARGET = 10.0;
+				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
バージョンあげた理由て何かあります？？

kazuteru commented on RssTab/model/Const.swift:
```diff
@@ -11,6 +11,7 @@ struct Const {
     static let ATNDBASELINK = "http://api.atnd.org/events/?keyword="
+    static let ERRORPAGE = "https://smartlog.jp/errorpage"
     
     static let TECHTITLE = "IT 科学ニュース"
```
大文字の区切りは_(アンスコ)を使うとよいかと
TECHTITLE → TECH_TITLE

kazuteru commented on RssTab/viewController/TabsViewController.swift:
```diff
@@ -8,6 +8,13 @@
 
 import UIKit
 
+enum ArticleViews {
```
ここは単数形ですね

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -26,10 +30,38 @@ class WebViewDetailViewController: UIViewController ,UIWebViewDelegate {
         title.textColor = UIColor.gray
         title.text = barTitle
         navigationItem.titleView = title
+        LoadWebView()
```
大文字始まりになってますね

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -53,11 +85,22 @@ class WebViewDetailViewController: UIViewController ,UIWebViewDelegate {
+        UIApplication.shared.isNetworkActivityIndicatorVisible = true
+        
+        // ボタンの有効性をチェック
+        self.goBackButton.isEnabled = self.webView.canGoBack
```
ここらへんはself不要そうですね

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -26,10 +30,38 @@ class WebViewDetailViewController: UIViewController ,UIWebViewDelegate {
+    @IBAction func goSafariButton(_ sender: Any) {
+        // その時開いているURLを取得
+        let nowWebViewURL = webView.url
+        UIApplication.shared.open(nowWebViewURL!, options: [:], completionHandler: nil)
```
ここは念の為 if letの形の方がいいですね
極力!は使わないように

kazuteru commented on RssTab/viewController/WebViewDetailViewController.swift:
```diff
@@ -7,17 +7,21 @@
 }
 
-class WebViewDetailViewController: UIViewController ,UIWebViewDelegate {
+class WebViewDetailViewController: UIViewController ,WKUIDelegate, WKNavigationDelegate {
```
UIViewController , → UIViewController, 
カンマのつけ方がおかしいですね

KazuGeoge committed:
文法の修正と列挙を使ったViewControllerのインスタンス化

文法の修正と列挙を使ったViewControllerのインスタンス化を行いました。
ご確認よろしくお願いします！
